premake5.lua

```diff
@@ -197,6 +197,10 @@
 			defines     { "LUA_USE_POSIX", "LUA_USE_DLOPEN" }
 			links       { "m" }
 
+		filter "system:haiku"
+			defines     { "LUA_USE_POSIX", "LUA_USE_DLOPEN", "_BSD_SOURCE" }
+			links       { "network", "bsd" }
```

Comment:
links { "network" } probably needs to go in the contrib/curl/premake5.lua under a haiku filter

Reply:
Hmm Solaris links with libsocket… besides, it seems you're building cURL as a static lib anyway.

Reply:
Yes, but if you build premake without those contribs, or if cURL ever gets made into a binmodule, this is not necessary for general premake usage.

Reply:
But for now cURL is used as a static lib, so you must link it here, since you can't link static libs to anything (it's just an archive).

Reply:
That is a good point, I had forgotten that detail about static libraries.

How about putting `links { "network" }` inside a `if not _OPTIONS["no-curl"] then ... end` check like on line 182?

Reply:
Wouldn't luasocket also need it?